### PR TITLE
API docs: point outdate link to current link

### PR DESCRIPTION
**Pull request by 0ekk**
#### What type of PR is this?
/kind cleanup
/kind documentation

#### What this PR does / why we need it:
From https://github.com/kubernetes/website/pull/45926#issuecomment-2067745121. These links are not available yet, point them to current locations.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
None
```

### CODE REVIEW

**Comment by sftim** [comment]
/remove-kind cleanup

This is a bug fix.

File: pkg/kubelet/apis/config/types.go [hunk 1]
```diff
@@ -374,12 +374,12 @@ type KubeletConfiguration struct {
 	// A set of ResourceName=ResourceQuantity (e.g. cpu=200m,memory=150G,ephemeral-storage=1G,pid=100) pairs
 	// that describe resources reserved for non-kubernetes components.
 	// Currently only cpu, memory and local ephemeral storage for root file system are supported.
-	// See http://kubernetes.io/docs/user-guide/compute-resources for more detail.
+	// See https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/ for more detail.
```
Comment (bart0sh) [inline pkg/kubelet/apis/config/types.go:377]: Would https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources be more appropriate here?

Comment (tu1h) [reply pkg/kubelet/apis/config/types.go]: Right

**Comment by bart0sh** [comment]
/cc @mrunalp @derekwaynecarr @dchen1107 
for approval

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm
/approve

**Comment by pacoxu** [comment]
broken links fix that is approved by sig-node
ping @kubernetes/api-approvers for approval
/lgtm

**Pull request merged**